src/house.cpp

```diff
@@ -132,13 +132,13 @@ AccessHouseLevel_t House::getHouseAccessLevel(const Player* player) const
 	return HOUSE_NOT_INVITED;
 }
 
-bool House::kickPlayer(Player* player, Player* target)
+bool House::kickPlayer(std::shared_ptr<Player> player, std::shared_ptr<Player> target)
 {
 	if (!target) {
 		return false;
 	}
 
-	HouseTile* houseTile = dynamic_cast<HouseTile*>(target->getTile());
+	auto houseTile = std::dynamic_pointer_cast<HouseTile>(target->getTile());
```

Comment:
To avoid using `dynamic_cast`, we could add `Tile::getHouseTile()` in `Tile`, just like `Thing::getTile()` in `Thing`.

Could this be `const auto`
```suggestion
	cont auto houseTile = std::dynamic_pointer_cast<HouseTile>(target->getTile());
```